### PR TITLE
Remove altinn-design-system package and clean up technical debt

### PR DESCRIPTION
**Pull request by mgunnerud**
## Description
- remove @altinn/altinn-design-system package
- upgrade packages
- remove deprecated as= prop from design system components
- use `<Link>` and `<Button>` from design system instead of native `<a>` and `<button>` (where applicable)
- refactor to remove `!important` from css

## Related Issue(s)
- #950 

## Verification
- [ ] **Your** code builds clean without any errors or warnings
- [ ] Manual testing done (required)
- [ ] Relevant automated test added (if you find this hard, leave it and we'll help out)
- [ ] All tests run green

## Documentation
- [ ] User documentation is updated with a separate linked PR in [altinn-studio-docs.](https://github.com/Altinn/altinn-studio-docs) (if applicable)

### CODE REVIEW

**Review by allinox — CHANGES_REQUESTED** [review]
Thank you for helping us clean up the code! 😄
Finally we are rid of the old design system! And you even fixed up a lot of outdated code too!

There are a few breaks and I've done my best to not them down whenever I found them, but there's a ton of great work here so some of the comments are just praise 😊

Have a great evening in the sunshine! ☀️

**Comment by allinox** [comment]
Another small note that I just realized: For future branches, please follow the [guidelines in our readme](https://github.com/Altinn/altinn-access-management-frontend?tab=readme-ov-file#naming-convention-on-branches) by at least prefixing the branch with feat/ or fix/ or chore/ etc.
We have an automatic labeller that labels PRs based on branch names you see, and our release notes are generated automatically as well based on those labels 😊

File: src/components/ActionBar/ActionBarHeader.module.css [hunk 1]
```diff
@@ -182,15 +177,6 @@
   justify-content: center;
 }
 
-.title,
-.subtitle {
-  color: var(--text-color) !important;
```
[the remark this answers is not in the document]
Comment (mgunnerud) [reply src/components/ActionBar/ActionBarHeader.module.css:187]: fixed, and looks like !important is no longer needed!

File: src/components/ActionBar/ActionBarHeader.module.css [hunk 2]
```diff
@@ -182,15 +177,6 @@
...
-}
-
-.title {
-  font-weight: 500 !important;
```
[the remark this answers is not in the document]
Comment (mgunnerud) [reply src/components/ActionBar/ActionBarHeader.module.css:191]: fixed, and looks like !important is no longer needed here either!

File: src/features/apiDelegation/components/OverviewPageContent/OverviewPageContent.tsx [hunk 3]
```diff
@@ -192,21 +191,21 @@ export const OverviewPageContent = ({
...
+          {t('api_delegation.api_panel_content')}{' '}
+          <Link
+            href='https://samarbeid.digdir.no/maskinporten/maskinporten/25'
+            target='_blank'
+            rel='noreferrer'
+          >
+            {t('common.maskinporten')}
+          </Link>
+        </Paragraph>
+      </Alert>
```
[the remark this answers is not in the document]
Comment (mgunnerud) [reply src/features/apiDelegation/components/OverviewPageContent/OverviewPageContent.tsx]: Changed the <Alert> to large and added a heading:

image

File: src/components/Filter/FilterButton.tsx [hunk 4]
```diff
@@ -53,27 +48,22 @@ export const FilterButton = forwardRef<HTMLButtonElement, FilterButtonProps>(
...
           className={classes.filterButton}
           onClick={onClick}
         >
-          {iconLeft && (
-            <SvgIcon
-              svgIconComponent={iconLeft}
-              className={classes.icon}
-            />
-          )}
+          <FilterIcon className={classes.icon} />
```
[the remark this answers is not in the document]
Comment (mgunnerud) [reply src/components/Filter/FilterButton.tsx]: Rolled back this change :D

File: src/components/Filter/OptionDisplay.tsx [hunk 5]
```diff
@@ -84,36 +84,22 @@ export const OptionDisplay = ({
...
-        <Paragraph
-          as={'label'}
-          className={classes.optionLabel}
-          size={compact ? 'small' : 'medium'}
-        >
-          {option.label}
-        </Paragraph>
-      </button>
+        {option.label}
+      </Checkbox>
```
[the remark this answers is not in the document]
Comment (mgunnerud) [reply src/components/Filter/OptionDisplay.tsx]: Rolled back this change as well 😅 turns out increasing the clickable height of the checkbox beyond 44px is not that easy, and I would rather not make very specific changes which makes assumptions of internal design system markup

File: src/features/apiDelegation/offered/ChooseOrgPage/ChooseOrgInfoPanel.tsx [hunk 6]
```diff
@@ -40,35 +39,31 @@ export const ChooseOrgInfoPanel = ({
     );
   } else if (!searchLoading && searchOrgNotExist) {
     return (
-      <Panel
-        variant={PanelVariant.Error}
-        showIcon={false}
+      <Alert
```
[the remark this answers is not in the document]
Comment (mgunnerud) [reply src/features/apiDelegation/offered/ChooseOrgPage/ChooseOrgInfoPanel.tsx:42]: Changed <Alert> size to large, and added heading back :D

image

**Review by allinox — APPROVED** [review]
Looks good! 😄

Added a cosmetic suggestion, but there's no show-stoppers as far as I can see.

File: src/features/apiDelegation/offered/ChooseOrgPage/ChooseOrgInfoPanel.tsx [hunk 7]
```diff
@@ -40,35 +39,45 @@ export const ChooseOrgInfoPanel = ({
...
-        variant={PanelVariant.Error}
-        showIcon={false}
-        title={String(t('api_delegation.buisness_search_notfound_title'))}
-        forceMobileLayout={true}
+      <Alert
+        severity='danger'
+        size='lg'
       >
-        <div>
+        <Heading
```
[the remark this answers is not in the document]
Comment (mgunnerud) [reply src/features/apiDelegation/offered/ChooseOrgPage/ChooseOrgInfoPanel.tsx:46]: True, I thought the original <Panel> was quite big as well 😅

image

I agree that two-sentence header is not optimal, I can move the second sentence to the body text!

[the remark this answers is not in the document]
Comment (allinox) [reply src/features/apiDelegation/offered/ChooseOrgPage/ChooseOrgInfoPanel.tsx:46]: Ouf! Yeah, that's not very pretty 😅
Let's give it a makeover 💄

**Pull request merged**